tests/cloud_run_csm_inbound_test.py

```diff
@@ -12,37 +12,56 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import logging
-import time
 
-from absl import flags
 from absl.testing import absltest
 
 from framework import xds_k8s_testcase
+from framework.test_cases import cloud_run_testcase
 
 logger = logging.getLogger(__name__)
-flags.adopt_module_key_flags(xds_k8s_testcase)
 
+_XdsTestServer = xds_k8s_testcase.XdsTestServer
+_XdsTestClient = xds_k8s_testcase.XdsTestClient
 
-class CloudRunCsmInboundTest(xds_k8s_testcase.RegularXdsKubernetesTestCase):
-    def test_gke_to_cloud_run_setup(self):
-        with self.subTest("0_started"):
-            time.sleep(2)
 
-        with self.subTest("1_executing"):
-            time.sleep(2)
+class CloudRunCsmInboundTest(cloud_run_testcase.CloudRunXdsKubernetesTestCase):
+    def test_gke_to_cloud_run(self):
+        with self.subTest("0_create_mesh"):
+            self.td.create_mesh()
 
-        with self.subTest("2_finished"):
-            time.sleep(2)
+        with self.subTest("1_start_cloud_run_test_server"):
+            test_server: _XdsTestServer = self.startTestServers()[0]
 
-    def test_cloud_run_to_cloud_run_setup(self):
-        with self.subTest("0_started"):
-            time.sleep(2)
+        with self.subTest("2_create_neg_serverless"):
+            self.td.create_neg_serverless(self.server_namespace)
 
-        with self.subTest("1_executing"):
-            time.sleep(2)
+        with self.subTest("3_create_backend_service"):
+            self.td.create_backend_service(
+                protocol=self.compute_v1.BackendServiceProtocol.HTTP2,
+            )
 
-        with self.subTest("2_finished"):
-            time.sleep(2)
+        with self.subTest("4_add_server_backends_to_backend_service"):
+            self.td.backend_service_add_cloudrun_backends(
+                [self.td.neg], self.region
```

Comment:
the arguments are wrong (despite working somehow)

> I0416 17:00:21.987738 140704465681344 cloud_run_mesh_manager.py:79] Adding backends to Backend Service stk-cr-backend-service-20250416-2359-dx898: [{'group': ComputeV1.GcpResource(name='stk-cr-grpc-neg-20250416-2359-dx898', url='...'), 'balancingMode': 'us-central1', 'capacityScaler': 1.0}]

Note `'balancingMode': 'us-central1',`

You don't need to pass the region anymore. In fact, you don't need to pass the backends too.
`self.td` is a resource manager, and its purpose is to track resources your test created.

It already knows what about `td.neg`. Note that `backend_service_patch_backends` doesn't have the backends in method signature too